### ISSUE TITLE
Feature Request: Real Container/Surface Support

### ISSUE THREAD

**Issue by ainslec** [ainslec/adventuron-issue-tracker#11]
Adventuron currently does not support real containers or supporters.

It should support containers which are openable, closable, lockable. 

Container accessibility should be supported.

Should be able to type

PUT XXX ON YYYY
PUT XXX IN YYYY
GET XXX OUT OF YYY
GET XXX FROM YYYY

LOOK ON XXXX
LOOK IN YYYY

Under is something under consideration.

System level container per object attributes will be part of this development. (per object attributes (traits) is already supported by adventuron, but in a non supported beta api).

**Comment by ainslec** [comment]
Pay attention to the following command:

`      : disambiguate_s1 category="present";`

Need to be able to disambiguate for all accessible items in the current location, including containers (and possibly subcontainers).

**Comment by ainslec** [comment]
API Thoughts:

```
// Implement in state model with a list of ids for the containment types

//
// chest
//    on       = [a,b,c]
//    in       = [d,e,f]
//    under    = [g,h,i]
//    atop     = [j,k,l]
//    carrying = [m,n.o]
//    part     = [p,q,r]

chest : object "a glass chest" { 
   container          = "true"    /* Can put items inside the object PUT LAMP IN CHEST, trait is _container */
   container_capacity = "5"       /* Container capacity and surface capacity should be different */
   transparent        = "true"    /* The container contents can be seen even if closed */
   weight             = "3"       /* weight of the container itself */
   surface            = "true"    /* if true then can put items atop of the item */
   has_space_under    = "false"   /* false is default anyway */
   lockable           = "true"    /* Lockable does not require key */
   openable           = "false"   /* The item can be open or _closed (open is default, so only _closed trait is necessary) */
   open               = "true"    /* Implies _openable */
   key                = "red_key" /* providing key implies lockable and openable, should implement per-node string key values. _container_key = red_key */
}
```

**Comment by ainslec** [comment]
Important, each object should have an attribute set for their containment state:

`_containment_type = in | on | under | part | carried | placed`

We should be able to interrogate this at runtime as follows:

```
: iterate "table"{
   // Iterates over items on or under the table (assuming the table has the _has_space_under trait set)
}
: iterate "table" filter -> ( (item_attribute "_containment_type") == "under") {
   // Iterates over items under the table
}
```
